figures/scripts/color_palettes.R

```diff
@@ -0,0 +1,164 @@
+# Create color palettes for each kind of data in OpenPBTA-analysis
+# C. Savonen for ALSF - CCDL 
+#
+# Usage: 
+#  Anywhere a plot is being made, source these TSV file and use the color palette for 
+#  each appropriate data type. 
+#
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Establish base dir
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# Output to palette directory
+output_dir <- file.path(root_dir, "figures", "palettes")
+
+### 1) A color for NA data
+na_color <- "#f1f1f1"
+
+### 2) Color palette for each histology group in short_histology.
+
+# Read in metadata file
+metadata <- readr::read_tsv(file.path(root_dir, "data", "pbta-histologies.tsv")) %>% 
+  dplyr::mutate(short_histology = as.character(tidyr::replace_na(short_histology, "none"))) %>% 
+  dplyr::arrange(short_histology)
+
+sort(summary(as.factor(metadata$short_histology)), decreasing = TRUE)
+# Get unique histology categories
+histologies <- unique(metadata$short_histology)
+
+# Create named list that can be used with dplyr::recode
+# These hex codes were retrieved from http://phrogz.net/css/distinct-colors.html
+# With the settings on default for 37 colors. 
+histology_color_key <- c("#f23d3d", "#731d1d", "#b38686", "#cc5c33", "#331c0d", 
+"#ffb380", "#b25f00", "#f2d6b6", "#736556", "#ffaa00", "#4c3d00", "#e2f200", 
+"#919926", "#d6f2b6", "#304d26", "#00f241", "#009929", "#698c7c", "#39e6c3", 
+"#005359", "#263233", "#00c2f2", "#40a6ff", "#406280", "#0044ff", "#00144d", 
+"#acbbe6", "#7373e6", "#3d0099", "#c200f2", "#917399", "#731d6d", "#f279da", 
+"#cc0052", "#994d6b", "#4d2636", "#ffbfd9")
+
+# Bring along histologies names
+names(histology_color_key) <- histologies
+
+# NA for histology we want to match the standard NA color
+histology_color_key[names(histology_color_key) == "none"] <- na_color
+
+# Add a column that has the color for each sample
+histology_sample_df <- metadata %>% 
+  dplyr::select(Kids_First_Biospecimen_ID, short_histology) %>%
+  dplyr::mutate(sample_color = dplyr::recode(short_histology, 
+                                             !!!histology_color_key)) %>% 
+  readr::write_tsv(file.path(output_dir, "histology_color_by_sample.tsv"))
+
+# Make a version with the color coding by group
+histology_group_df <- data.frame(histology_color_key) %>% 
+  readr::write_tsv(file.path(output_dir, "histology_color_by_group.tsv"))
```

Comment:
Taking a look at the output of these two steps – I don't think you want to be in a situation where if a `short_histology` label changes the color palette that people are using for their biospecimens is out of sync. So with that in mind, I suggest we do the following:

* Remove `histology_color_by_sample.tsv` entirely
* Make `histology_color_by_group.tsv` more similar to the other palettes where the first column is the "name" of the color (here it will be `short_histology` instead). 
* Add an example where you use a `join` to make the equivalent of `histology_color_by_sample.tsv` to the documentation. Basically, show people how to use `histology_color_by_group.tsv` in their analyses.

Reply:
Okay. I like this plan. I had both by group and by sample because I wanted to make it as easy as possible for people to implement the color palette, however, I see this could definitely get out of sync easily.